grants_tagger/train.py

```diff
@@ -223,14 +225,44 @@ def train_and_evaluate(
             label_binarizer=label_binarizer,
             from_same_distribution=from_same_distribution
         )
-        model.fit(X_train, Y_train)
+        if y_batch_size:
+            vectorizer = model.steps[0][1]
+            classifier = model.steps[1][1]
+
+            Path(model_path).mkdir(exist_ok=True)
+            print("Fitting vectorizer")
+            vectorizer.fit(X_train)
+            with open(f"{model_path}/vectorizer.pkl", "wb") as f:
+                f.write(pickle.dumps(vectorizer))
+            print("Training model")
+            for tag_i in range(0, Y_train.shape[1], y_batch_size):
+                print(tag_i)
+                X_train_vec = vectorizer.transform(X_train)
+                classifier.fit(X_train_vec, Y_train[:,tag_i:tag_i+512]) # assuming that fit clears previous params
+                with open(f"{model_path}/{tag_i}.pkl", "wb") as f:
+                    f.write(pickle.dumps(classifier))
+        else:
+            model.fit(X_train, Y_train)
 
     if threshold:
-        Y_pred_prob = model.predict_proba(X_test)
+        if y_batch_size:
+            pass # to be implemented
+        else:
+            Y_pred_prob = model.predict_proba(X_test)
         Y_pred_test = Y_pred_prob > threshold
     else:
-        Y_pred_test = model.predict(X_test)
-    #    Y_pred_train = model.predict(X_train)
+        if Y_test.shape[1] > 512:
+            Y_pred_test = []
+            for tag_i in range(0, Y_test.shape[1], 512):
```

Comment:
Same as before - where did 512 come from? Is it a default batch size?

Reply:
same treatment as above

Reply:
Resolved by 2701cf6ab0b2232256378fe5a327d940f242c36e